Library/Homebrew/formula.rb

```diff
@@ -2608,6 +2609,19 @@ def dependencies_hash
     hash
   end
 
+  def dependencies_list(spec_symbol)
+    return if spec_symbol != :stable && spec_symbol != :head
+
+    send(spec_symbol)&.declared_deps&.each_with_object({}) do |dep, dep_hash|
+      next if dep.implicit? # Remove all implicit deps
```

Comment:
Would be good to say why they are being removed rather than that they are

Reply:
This comment was copied from the old logic. I'll do some digging and update it.

Reply:
It seems like implicit dependencies are really only needed when building from source and are used when downloading and unpacking files. See `DependencyCollector` for more details.